### PR TITLE
neighbor advertiser script

### PR DESCRIPTION
**Pull request by trzhang-msft**
Neighbor Advertiser Script

Set:
- set address mapping <vlan_mac : vlan_ip> on Ferret server
- set mirror encap tunnel (arp)
- set vxlan decap tunnel

Reset:
- reset mirror encap tunnel (arp)
- reset vxlan decap tunnel

### CODE REVIEW

**Comment by pavel-shirshov** [comment]
Updated version of: https://github.com/Azure/sonic-utilities/pull/351

File: scripts/neighbor_advertiser [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+########################################
+#
+# Neighbor Advertiser
+#
+########################################
+
+
+import os
+import json
+import socket
```
[the remark this answers is not in the document]
Comment (trzhang-msft) [reply scripts/neighbor_advertiser]: fixed

File: scripts/neighbor_advertiser [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+import subprocess
```
[the remark this answers is not in the document]
Comment (trzhang-msft) [reply scripts/neighbor_advertiser]: fixed

File: scripts/neighbor_advertiser [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+from swsssdk import ConfigDBConnector
+from swsssdk import SonicV2Connector
+
+
+#
+# Config
+#
+
+DEFAULT_DURATION = 300
+DEFAULT_DECODING = 'utf-8'
```
[the remark this answers is not in the document]
Comment (trzhang-msft) [reply scripts/neighbor_advertiser]: fixed

File: scripts/neighbor_advertiser [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+            log_warning('Cannot find mac addr of vlan interface {}'.format(vlan_interface))
+            continue
+
+        ipv4_mappings = []
+        for addr in ipv4_addr:
+            mapping = {
+                'ipAddr': addr,
+                'macAddr': mac_addr
+            }
+
```
[the remark this answers is not in the document]
Comment (trzhang-msft) [reply scripts/neighbor_advertiser]: updated

File: scripts/neighbor_advertiser [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+    return slice_obj
+
+
+def post_neighbor_advertiser_slice(ferret_service_vip):
+    request_slice = construct_neighbor_advertiser_slice()
+    save_as_json(request_slice, NEIGHBOR_ADVERTISER_REQUEST_SLICE_PATH)
+
+    url = 'http://{}:85{}{}'.format(ferret_service_vip, FERRET_NEIGHBOR_ADVERTISER_API_PREFIX, get_switch_name())
+    response = requests.post(url, json = request_slice)
```
[the remark this answers is not in the document]
Comment (trzhang-msft) [reply scripts/neighbor_advertiser]: code for catching exceptions is there

File: scripts/neighbor_advertiser [hunk 6]
```diff
@@ -0,0 +1,456 @@
...
+    ferret_server_ipv4_addr = ''
+
+    if response.ok:
+        neighbor_advertiser_configuration = json.loads(response.content)
+        save_as_json(neighbor_advertiser_configuration, NEIGHBOR_ADVERTISER_RESPONSE_CONFIG_PATH)
+        ferret_server_ipv4_addr = neighbor_advertiser_configuration['ipv4Addr']
+        log_info('Successfully set up neighbor advertiser slice')
+    else:
+        log_error('Failed to set up neighbor advertiser slice')
+        raise Exception('Phase: post neighbor advertiser slice')
```
[the remark this answers is not in the document]
Comment (trzhang-msft) [reply scripts/neighbor_advertiser]: thx, addressed

File: scripts/neighbor_advertiser [hunk 7]
```diff
@@ -0,0 +1,456 @@
...
+        reset_vxlan_tunnel()
+
+    return true
+
+
+if __name__ == '__main__':
+    try:
+        main()
+    except Exception as e:
+        log_error('! [Failure] {}'.format(e))
```
[the remark this answers is not in the document]
Comment (trzhang-msft) [reply scripts/neighbor_advertiser]: thx, added

File: scripts/neighbor_advertiser [hunk 8]
```diff
@@ -0,0 +1,456 @@
...
+
+            ipv4_mappings.append(mapping)
+
+        ipv6_mappings = []
+        for addr in ipv6_addr:
+            mapping = {
+                'ipAddr': addr,
+                'macAddr': mac_addr
+            }
+
```
[the remark this answers is not in the document]
Comment (trzhang-msft) [reply scripts/neighbor_advertiser]: updated

File: scripts/neighbor_advertiser [hunk 9]
```diff
@@ -0,0 +1,456 @@
...
+#
+# Syslog functions
+#
+
+def log_info(msg):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
```
[the remark this answers is not in the document]
Comment (trzhang-msft) [reply scripts/neighbor_advertiser]: i think it is fine, this way we can use the func as primitive to log info / error on the fly

File: scripts/neighbor_advertiser [hunk 10]
```diff
@@ -0,0 +1,456 @@
...
+    log_info('Finish resetting vxlan tunnel')
+
+
+#
+# Main function
+#
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-s', dest='vip', metavar='vip', type = str, required = False, help = 'ferret service vip')
```
[the remark this answers is not in the document]
Comment (trzhang-msft) [reply scripts/neighbor_advertiser]: updated, now in reset phase, pass an empty string

**Pull request merged**